compiler/rustc_builtin_macros/src/proc_macro_harness.rs

```diff
@@ -329,22 +333,8 @@ fn mk_decls(cx: &mut ExtCtxt<'_>, macros: &[ProcMacro]) -> P<ast::Item> {
                         ],
                     )
                 }
-                ProcMacro::Def(ca) => {
-                    cx.resolver.declare_proc_macro(ca.id);
-                    let ident = match ca.def_type {
-                        ProcMacroDefType::Attr => attr,
-                        ProcMacroDefType::Bang => bang,
-                    };
-
-                    cx.expr_call(
-                        span,
-                        proc_macro_ty_method_path(cx, ident),
-                        vec![
-                            cx.expr_str(ca.span, ca.function_name.name),
-                            local_path(cx, ca.span, ca.function_name),
-                        ],
-                    )
-                }
+                ProcMacro::Attr(ca) => attr_or_bang(cx, &ca, attr),
+                ProcMacro::Bang(ca) => attr_or_bang(cx, &ca, bang),
```

Comment:
You can simplify this as
```suggestion
                ProcMacro::Bang(ca) | ProcMacro::Attr(ca) => attr_or_bang(cx, &ca, bang),
```
which in turn allows you to remove the closure and keep the original inline code, which is less churn.

Reply:
Alas, no. The `Attr` match arm uses `attr` from above, and the `Bang` match arm uses `bang` from above.